testsuite/smoke-test/src/rosetta.rs

```diff
@@ -54,36 +63,111 @@ pub async fn setup_test(
 }
 
 #[tokio::test]
-#[ignore]
+async fn test_network() {
+    let (swarm, _, _, rosetta_client) = setup_test(1, 1).await;
+    let chain_id = swarm.chain_id();
+
+    // We only support one network, this network
+    let networks = rosetta_client.network_list().await.unwrap();
+    assert_eq!(1, networks.network_identifiers.len());
+    let network_id = networks.network_identifiers.first().unwrap();
+    assert_eq!(BLOCKCHAIN, network_id.blockchain);
+    assert_eq!(chain_id.to_string(), network_id.network);
+
+    let request = NetworkRequest {
+        network_identifier: NetworkIdentifier::from(chain_id),
+    };
+    let options = rosetta_client.network_options(&request).await.unwrap();
+    assert_eq!(ROSETTA_VERSION, options.version.rosetta_version);
+
+    // TODO: Check other options
+
+    let request = NetworkRequest {
+        network_identifier: NetworkIdentifier::from(chain_id),
+    };
+    let status = rosetta_client.network_status(&request).await.unwrap();
+    assert!(status.current_block_identifier.index > 0);
```

Comment:
Is there a potential race here, or will the block identifier always start from > 0 with the setup we do? As in, couldn't the current block be the genesis block?

Reply:
Yes, I think there is a potential race for 0, but the first thing done by consensus is a reconfig, so it's highly unlikely.